src/shared/components/SharePost.tsx

```diff
@@ -8,27 +8,38 @@ import {
   FormControl,
   InputBase,
   Typography,
+  Input,
 } from "@mui/material";
-import { useForm } from "../hooks";
+import { useForm, useImageInput } from "../hooks";
+import { uploadMedia } from "shared/utils/cloudinaryUtil";
 import { BaseCreatePostmodel } from "../types/SharePostTypes";
+import axios from "axios";
 
 type SharePostProps = {
-  readonly profilePic: string;
-  readonly sx: Record<string, string>;
-  readonly isPost: boolean;
-  readonly onClick: (value: any) => void;
+  readonly profilePic?: string;
+  theme: any;
 };
 
-const SharePost = ({ profilePic, sx, onClick, isPost }: SharePostProps) => {
-  const { form, handleChange } = useForm<BaseCreatePostmodel>({
+const SharePost = ({ profilePic, theme }: SharePostProps) => {
+  const { form, handleChange, resetForm } = useForm<BaseCreatePostmodel>({
     media: "",
     content: "",
     categoryName: "",
   });
-  const { media, content, categoryName } = form;
+
+  const { content, categoryName } = form;
+  const { onSelectFile, preview, onRemove, selectedFile } = useImageInput();
+  const endPoint = process.env.REACT_APP_BASE_URL + "/posts";
```

Comment:
I would move this inside of handleSubmit.

Reply:
gotcha